### PR TITLE
cpu/stm32x: unified UART driver implementations

### PR DESCRIPTION
**Pull request by haukepetersen**
Now finally also a shared UART implementation for all the `stm32x` CPUs.

~~This PR shares two commits with #6186 (centralizing of periph_apb_clk).~~

NOTE: Currently I dropped the DMA support (offered previously by `stm32f2` and `stm32f4`). Especially when used with STDIO and then using printf/puts from ISR context, this was not the best idea to enable by default anyway... I plan to re-add this functionality again in a separate PR (together with a stm32x wide concept for DMA handling).

### CODE REVIEW

**Comment by haukepetersen** [comment]
restarted Murdock (as he was confused by the 'waiting for other PR' label)

File: boards/f4vi1/include/periph_conf.h [hunk 1]
```diff
@@ -78,28 +78,29 @@ static const timer_conf_t timer_config[] = {
 /** @} */
 
 /**
- * @name UART configuration
+ * @brief   UART configuration
```
Comment (A-Paul) [inline boards/f4vi1/include/periph_conf.h]: I noticed this also in other PRs. What is the deal with this extra space?

Comment (miri64) [inline boards/f4vi1/include/periph_conf.h]: Also, check if this parses still correctly. Without the name the sectioning in the doc often isn't correct anymore (and the brief of the group is added to the first element)

Comment (haukepetersen) [inline boards/f4vi1/include/periph_conf.h]: Actually, I have never spend much thought on this -> I have been using brief for everything for the last 2 years I think... 

About the spaces: they are there to have a unified indention in a doc comment, so if you use e.g. return, note, and brief in one block, their texts all start at the same col.

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
rebased again

**Comment by haukepetersen** [comment]
fixed some issue with the c++ compiler and rebased.

**Comment by aabadie** [comment]
needs rebase

**Comment by aabadie** [comment]
Successfully tested on 3 nucleo boards : nucleo-f030, nucleo-f070 and nucleo-f401

**Comment by haukepetersen** [comment]
@kYc0o: could you be so nice and test on the `iotlab-m3`? Thx

**Comment by aabadie** [comment]
@kYc0o, @haukepetersen, just tested iotlab-m3 : the shell works.

Also tested all nucleo boards except f091, f207 and f334 => OK

I tend to give an ACK.

**Comment by haukepetersen** [comment]
Me too :-)

**Pull request merged**